public/app/app.ts

```diff
@@ -380,4 +381,49 @@ function reportMetricPerformanceMark(metricName: string, prefix = '', suffix = '
   }
 }
 
+function handleRedirectTo(): void {
```

Comment:
Why the change to read this on app startup, rather than just issue the 'redirect' to the final destination immediately after login?

Reply:
>issue the 'redirect' to the final destination immediately after login?

Could you please elaborate?

For more context: there are cases when the redirect url is lost during the requests (OAuth and SAML flows), and previously those backend handlers handled the redirect after login, but with this change (removing the `redirect_to` cookie and using sessionStorage instead) the backend doesn't know where to redirect the user after OAuth/SAML login. (Although we could potentially send the `state` for both SAML and OAuth I decided to implement a solution without changing the `state` params of those)

Reply:
Ahhh, I understand. I was not considering oauth flows where the user gets logged in off of grafana, and is redirected back to us.